scripts/get_graphids.py

```diff
@@ -0,0 +1,160 @@
+from taskcluster.async import Queue
+
+
+HG_URL = 'https://hg.mozilla.org/'
+
+REPOS = ('releases/mozilla-beta', 'releases/mozilla-release',
+         'releases/mozilla-esr52', 'mozilla-central')
+
+""" http://mozilla-version-control-tools.readthedocs.io/en/latest/hgmo/pushlog.html?highlight=json-pushes#json-pushes-command """  # noqa
+JSON_PUSHES = 'json-pushes?changeset={rev}'
+
+ACTION_INDEX = 'gecko.v2.{repo}.pushlog-id.{pushid}.actions'
+
+
+async def get_changeset_data(revision, session):
+    for repo in REPOS:
+        url = posixpath.join(HG_URL, repo, JSON_PUSHES.format(rev=revision))
+        async with async_timeout.timeout(10):
+            async with session.get(url) as response:
+                if response.status == 404:
+                    continue
+                response.raise_for_status()
+
+                index_repo = repo[repo.rfind('/') + 1:]
+                data = await response.json()
+                pushlog_id = list(data.keys())[0]
+                changeset = data[pushlog_id]['changesets'][-1]
+                return (changeset, index_repo, pushlog_id)
+
+
+async def get_action_tasks(session, pushlog_id, index_repo):
+    async with async_timeout.timeout(100):
+        index_string = ACTION_INDEX.format(pushid=pushlog_id, repo=index_repo)
+        index = Index(session=session)
+        data = await index.listTasks(index_string, dict(limit=100))
+        tasks = [t['taskId'] for t in data['tasks']]
+        return tasks
+
+
+async def get_action_task_details(session, taskid):
+    async with async_timeout.timeout(100):
+        queue = Queue(session=session)
+        task = await queue.task(taskid)
+        return dict(taskid=taskid,
+                    name=task['extra']['action']['name'],
+                    buildnum=task['extra']['action']['context']['input']['build_number'],
+                    flavor=task['extra']['action']['context']['input']['release_promotion_flavor'],
+                    ci=task['taskGroupId'])
+
+
+def output_graphs(tasks, format='human'):
+    if format == 'human':
+        return output_graphs_human(tasks)
+    if format == 'export':
+        return output_graphs_export(tasks)
+    raise NotImplementedError("format {} is not yet implemented".format(format))
+
+
+def output_graphs_human(tasks):
+    for product in ['devedition', 'firefox', 'fennec']:
+        product_tasks = [t for t in tasks if product in t['flavor']]
+        if not product_tasks:
+            continue
+        buildnumbers = set([t['buildnum'] for t in product_tasks])
+        print("|{:-^111}|".format(product.title()))
+        print("|  {:<8} | {:^22} | {:^22} | {:^22} | {:^22} |".format(
+            'buildN', 'CI', 'promote', 'push', 'ship'))
+        print("|{:-^111}|".format(''))
+        for bn in sorted(buildnumbers, reverse=True):
+            for ci in set([t['ci'] for t in product_tasks if t['buildnum'] == bn]):
+                promote = [t['taskid'] for t in product_tasks
```

Comment:
we have this pattern in a couple places. We may want to have a `KNOWN_PHASES` tuple/dict and loop through that. If we knew the repo, we could even download the appropriate tree's [release_promotion.py](https://hg.mozilla.org/releases/mozilla-beta/file/tip/taskcluster/taskgraph/actions/release_promotion.py#l23) and use it to determine what the relpro flavor names are.

Reply:
I agree this pattern could be improved, I'm going to leave that for 'next releaseduty' at this point.